### PR TITLE
Introduce dev_api for core component

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Introduce dev_api for core library

### Tickets:
 - NA

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/intel_cpu/CMakeLists.txt [hunk 1]
```diff
@@ -66,30 +66,31 @@ ie_add_api_validator_post_build_step(TARGET ${TARGET_NAME})
 
 #  add test object library
 
-add_library(${TARGET_NAME}_obj OBJECT ${SOURCES} ${HEADERS})
-link_system_libraries(${TARGET_NAME}_obj PUBLIC mkldnn)
+if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/CMakeLists.txt:69]: `BUILD_SHARED_LIBS` is always defined

File: src/tests/unit/cpu/CMakeLists.txt [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 set(TARGET_NAME cpuUnitTests)
 
+if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+    set (OBJ_LIB $<TARGET_OBJECTS:ov_intel_cpu_plugin_obj>)
```
Comment (ilya-lavrenov) [inline src/tests/unit/cpu/CMakeLists.txt:8]: the same

Comment (ilyachur) [reply src/tests/unit/cpu/CMakeLists.txt:8]: Fixed

File: src/plugins/intel_cpu/CMakeLists.txt [hunk 3]
```diff
@@ -66,30 +66,31 @@ ie_add_api_validator_post_build_step(TARGET ${TARGET_NAME})
...
+        )
 
-# LTO
+    set_target_properties(${TARGET_NAME}_obj PROPERTIES EXCLUDE_FROM_ALL ON)
 
-set_target_properties(${TARGET_NAME} ${TARGET_NAME}_obj
-                      PROPERTIES INTERPROCEDURAL_OPTIMIZATION_RELEASE ${ENABLE_LTO})
+    # LTO
+    set_target_properties(${TARGET_NAME} ${TARGET_NAME}_obj
+        PROPERTIES INTERPROCEDURAL_OPTIMIZATION_RELEASE ${ENABLE_LTO})
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/CMakeLists.txt:95]: why LTO is enabled only for shared libs for CPU? 
Before this change, it was enabled always

Comment (ilyachur) [reply src/plugins/intel_cpu/CMakeLists.txt:95]: We use this library only for dynamic build

Comment (ilya-lavrenov) [reply src/plugins/intel_cpu/CMakeLists.txt:95]: but LTO should be enabled for main target `${TARGET_NAME}` even for static build

**Pull request merged**